examples/src/main/python/ml/estimator_transformer_param_example.py

```diff
@@ -54,7 +56,7 @@
     print(model1.extractParamMap())
 
     # We may alternatively specify parameters using a Python dictionary as a paramMap
-    paramMap = {lr.maxIter: 20}
+    paramMap: Dict[Param, Any] = {lr.maxIter: 20}
```

Comment:
@zero323, is it because mypy complains? We can exclude the example files from the mypy check for the time being I guess rather than adding the types into the codes for now.

Reply:
It is. In such cases  Mypy infers the type of `paramMap` to be `Dict[Param, int]` and will fail on subsequent update:

```python
paramMap.update({lr.regParam: 0.1, lr.threshold: 0.55}) 
```

In general I'd prefer to keep tests against examples as these are the biggest chunks of "real-life" code (short of docstrings, but these create way more issues)  that we have and nicely highlight many possible problems.

In `pyspark-stubs` I simply clone the repo and [patch examples](https://github.com/zero323/pyspark-stubs/blob/faad51cfd4b8971cda889f79c6857dd28ad62078/.travis.yml#L11-L12) before tests.

Reply:
I don't object to having the type hints in the examples but I guess we should type everything in examples if we want. We could discuss and do separately - we'd likely have to file multiple followup JIRAs under SPARK-32681.

I was thinking it's good to port the PySpark type hints alone first with minimized diff to other codes in Spark in this JIRA/PR.

Reply:
I guess we could just `type: ignore` such cases - it would get most (or at least some) of the benefits, while by unobtrusive.